### PR TITLE
reptyr: enable on 32-bit ARM Linux and x86 FreeBSD

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Reptyr should not restricted to specific architectures. I tested it and it works fine on armv7l.

###### Things done

Made the platforms less restrictive, allowing it to run on all Linux architectures. MacOS is not supported.

cc @7c6f434c

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build reptyr

**Comment by lopsided98** [comment]
Oops, didn't test this on aarch64. I guess it only supports 32-bit ARM.

**Comment by lopsided98** [comment]
I updated the platforms to support 32-bit ARM, and added freebsd as well because it is supposedly supported, although I don't have a machine to test it with.

I also updated the description to the current GitHub description, which reflects that fact that it is no longer only supported on Linux.

**Comment by 7c6f434c** [comment]
Would `with stdenv.lib.platforms; linux ++ freebsd` work then?

**Comment by c0bw3b** [comment]
`platforms.linux` would include aarch64-linux and it seems it fails to build on ARMv8

File: pkgs/os-specific/linux/reptyr/default.nix [hunk 1]
```diff
@@ -20,9 +20,14 @@ stdenv.mkDerivation rec {
 
   makeFlags = ["PREFIX=$(out)"];
   meta = {
```
Comment (c0bw3b) [inline pkgs/os-specific/linux/reptyr/default.nix]: Also add `meta.homepage`

Comment (lopsided98) [reply pkgs/os-specific/linux/reptyr/default.nix]: Done.

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build reptyr

**Pull request merged**